zenodo/modules/records/jsonschemas/records/file-v1.0.0.json

```diff
@@ -3,51 +3,71 @@
   "id": "http://zenodo.org/schemas/records/file-v1.0.0.json",
   "properties": {
     "_buckets": {
+      "type": "object",
       "properties": {
         "deposit": {
-          "description": "Deposit bucket ID.",
+          "description": "UUID of the deposit bucket.",
           "type": "string"
         },
         "record": {
-          "description": "Record bucket ID.",
+          "description": "UUID of the record bucket.",
           "type": "string"
         }
-      },
-      "type": "object"
+      }
     },
     "_files": {
+      "description": "List of files attached to the record.",
       "items": {
-        "description": "Describe information needed for files in records.",
+        "description": "Describes the information of a single file in the record.",
         "properties": {
           "bucket": {
+            "description": "UUID of the bucket to which this file is assigned.",
             "type": "string"
           },
           "checksum": {
+            "description": "Checksum the file. Starts with '<algorithm>:' prefix, e.g.: 'md5:1234abcd...'",
+            "type": "string"
+          },
+          "file_id": {
+            "description": "UUID of the FileInstance object.",
             "type": "string"
           },
           "key": {
+            "description": "Key (filename) of the file.",
             "type": "string"
           },
           "previewer": {
             "description": "Identifier for previewer needed to preview this file.",
             "type": "string"
           },
           "size": {
+            "description": "Size of the file in bytes.",
             "type": "integer"
           },
           "type": {
             "description": "File type extension.",
             "type": "string"
           },
           "version_id": {
+            "description": "UUID of the ObjectVersion object.",
             "type": "string"
           }
         },
+        "required": [
+          "bucket",
+          "checksum",
+          "file_id",
```

Comment:
Could this lead on issues (eg. merge conflicts) with re-publishing old records that are missing some of the fields?

Reply:
I don't think so. In general merging of old records takes the record's metadata at the revision specified by the deposit (_deposit.pid_revision_id), computes a diff and applies the diff on the newest record metadata. Conflicts occur if the newest record's metadata conflicts with the diff of two jsons, but in this case I don't think it's possible, since somehow user would have to edit the deposits "_files"